### PR TITLE
Fix erors caused by -Wextra -Wall -Wpedantic -Werror

### PR DESCRIPTION
**Pull request by paceholder**
I fixed a couple of compilation problems mainly caused by unstrict implicit type convertions.

### CODE REVIEW

File: test/src/sw/redis++/string_cmds_test.hpp [hunk 1]
```diff
@@ -148,7 +148,7 @@ void StringCmdTest<RedisInstance>::_test_bit() {
 
     // dest_key -> 11101111
     v = _redis.get(dest_key);
-    REDIS_ASSERT(v && *v == std::string(1, 0xEF), "failed to test bitop");
+    REDIS_ASSERT(v && *v == std::string(1, '\xEF'), "failed to test bitop");
```
Comment (paceholder) [inline test/src/sw/redis++/string_cmds_test.hpp:151]: `0XEF` is by default a value of `int`, it cannot be automatically converted to `char` without losing integer precision.

File: test/src/sw/redis++/zset_cmds_test.hpp [hunk 2]
```diff
@@ -206,7 +206,7 @@ void ZSetCmdTest<RedisInstance>::_test_range() {
             std::back_inserter(members));
     REDIS_ASSERT(members.size() == sReversedKeys.size()-1,
                  "failed to test zrevrangebyscore (size)");
-    for (int i=0; i<members.size(); i++) {
+    for (size_t i=0; i<members.size(); i++) {
```
Comment (paceholder) [inline test/src/sw/redis++/zset_cmds_test.hpp:209]: standard containers always have `std::size_t` for their sizes

File: test/src/sw/redis++/zset_cmds_test.hpp [hunk 3]
```diff
@@ -134,7 +134,7 @@ void ZSetCmdTest<RedisInstance>::_test_range() {
 
     _redis.zadd(key, s.begin(), s.end());
 
-    REDIS_ASSERT(_redis.zcount(key, UnboundedInterval<double>{}) == s.size(),
+    REDIS_ASSERT(_redis.zcount(key, UnboundedInterval<double>{}) == static_cast<long long int>(s.size()),
```
Comment (paceholder) [inline test/src/sw/redis++/zset_cmds_test.hpp:137]: `zcount` returns signed integer, a container length is an unsigned integer. Assuming the length of the container is relatively small in this synthetic test, we can use this simple cast.

**Comment by sewenew** [comment]
@paceholder  Thanks for your work!

Regards

**Pull request merged**